src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Semantic/SemanticTokenTestBase.cs

```diff
@@ -67,6 +67,22 @@ internal void AssertSemanticTokensMatchesBaseline(IEnumerable<int>? actualSemant
             var semanticIntStr = semanticFile.ReadAllText();
             var semanticArray = ParseSemanticBaseline(semanticIntStr);
 
+            if (semanticArray is null && actual is null)
+            {
+                return;
+            }
+            else if (semanticArray is null || actual is null)
+            {
+                Assert.False(true, $"Expected: {semanticArray}; Actual: {actual}");
+            }
+
+            for (var i = 0; i < Math.Min(semanticArray!.Length, actual!.Length); i += 5)
```

Comment:
So was the Assert.Equal on line 86 not doing the deed before?

Reply:
I [addressed this here](https://github.com/dotnet/aspnetcore-tooling/pull/2907#discussion_r554086775) if I understand the question.

Reply:
Ahhh ok you added clarity in the comment. if that check is purely to check lengths it might be more clear tosimply add the length check